pkg/api/types/validation/recipient_test.go

```diff
@@ -0,0 +1,84 @@
+package validation
+
+import (
+	"testing"
+
+	"github.com/nrc-no/core/pkg/api/types"
+	"github.com/nrc-no/core/pkg/validation"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_validateRecipientFormHasSingleKeyField(t *testing.T) {
+	tests := []struct {
+		name           string
+		fields         types.FieldDefinitions
+		path           *validation.Path
+		wantField      *types.FieldDefinition
+		wantFieldIndex int
+		wantErrs       validation.ErrorList
+	}{
+		{
+			name:           "without key field",
+			fields:         []*types.FieldDefinition{},
+			path:           validation.NewPath("fields"),
+			wantField:      nil,
+			wantFieldIndex: -1,
+			wantErrs:       nil,
+		},
+		{
+			name: "with single key field",
+			fields: []*types.FieldDefinition{
+				{
+					Key: true,
+					FieldType: types.FieldType{
+						Text: &types.FieldTypeText{},
+					},
+				},
+			},
+			path: validation.NewPath("fields"),
+			wantField: &types.FieldDefinition{
+				Key: true,
+				FieldType: types.FieldType{
+					Text: &types.FieldTypeText{},
+				},
+			},
+			wantFieldIndex: 0,
+			wantErrs:       nil,
+		},
+		{
+			name: "with multiple key field",
+			fields: []*types.FieldDefinition{
```

Comment:
could we also add a test for testing if that is valid for non-recipients? I don't see where this example form is defined as a recipient form, so this might as well be valid

Reply:
I`m not sure how to write this test.. 
But this validation is skipped for non-recipient forms. Can you clarify what you would change ?

```
// ValidateRecipientForm validates recipient forms
func ValidateRecipientForm(form *types.FormDefinition) validation.ErrorList {
	var result validation.ErrorList
	if form.Type != types.RecipientFormType {
		return result
	}
         ...
```

Reply:
okay, true, that doesn't make sense then. but we could add a test that ValidateRecipientForm really skips the validation/returns an empty error list when the form is not a recipient form?

Reply:
Sure, will do !

Reply:
@ludydoo it's a detail, but in the test you added, you test ValidateForm instead of ValidateRecipientForm, is there a reason for that?

Reply:
@internetti I wanted the test to also validate that the ValidateRecipientForm was called by the main "ValidateForm"